### PR TITLE
added redux for register, and send information to database

### PR DESCRIPTION
**Pull request by WenxuC**

### CODE REVIEW

File: client/src/actions/auth.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+		};
+		const body = JSON.stringify({ userName, email, password });
+		try {
+			const res = await axios.post('/app/register', body, config);
+
+			dispatch({
+				type: REGISTER_COMPLETE,
+				payload: res.data,
+			});
+		} catch (err) {
```
Comment (channnx) [inline client/src/actions/auth.js:22]: What's the difference between err and error, are they having different value "?

File: client/src/components/Signup.js [hunk 2]
```diff
@@ -18,15 +19,15 @@ import { createTheme, ThemeProvider } from '@mui/material/styles';
...
+const Signup = ({ setAlert, signUp }) => {
 	const [formData, setFormData] = useState({
 		userName: '',
 		email: '',
 		password: '',
 		password_confirm: '',
 	});
 
-	const { name, email, password, password_confirm } = formData;
+	const { userName, email, password, password_confirm } = formData;
```
Comment (channnx) [inline client/src/components/Signup.js:30]: Do we need to change in db models too? It will send back to db, if it doesn't match will it return error ?

**Review by channnx — APPROVED** [review]

File: client/src/actions/auth.js (hunk 1 above)
Comment (WenxuC) [reply client/src/actions/auth.js:22]: im catching the error in err and then saving it in errors

Comment (channnx) [reply client/src/actions/auth.js:22]: Just catch the err and console.log(err)

Comment (WenxuC) [reply client/src/actions/auth.js:22]: there are multiple errors

Comment (WenxuC) [reply client/src/actions/auth.js:22]: i tried console.log(err) now all the errors wont show. thanks for your input.

File: client/src/components/Signup.js (hunk 2 above)
Comment (WenxuC) [reply client/src/components/Signup.js:30]: if you look into the db model, you can see the property is userName

**Pull request merged**